test/parallel/test-http-response-add-header-after-sent.js

```diff
@@ -0,0 +1,21 @@
+'use strict';
+const common = require('../common');
+const assert = require('assert');
+const http = require('http');
+
+const server = http.createServer((req, res) => {
+  assert.doesNotThrow(() => {
+    res.setHeader('header1', 1);
+  });
+  res.write('abc')
+  assert.throws(() => {
+    res.setHeader('header2', 2);
+  }, Error, 'setHeader after write should throw');
+  res.end()
```

Comment:
Well, if that's considered a breaking change then I'm also for checking the error message :-)
Thank you for the clarification. I will fix it.